nixos/modules/services/networking/sabnzbd/default.nix

```
@@ -0,0 +1,605 @@
+                '';
+                default = null;
+                example = literalExpression "\${config.acme.certs.\${domain}.directory}/fullchain.pem";
+              };
+              https_key = mkOption {
+                type = types.nullOr types.path;
+                description = ''
+                  Path to the TLS key for the web UI. If not set and
+                  https is enabled, a self-signed certificate will be
+                  generated
+                '';
+                default = null;
+                example = literalExpression "\${config.acme.certs.\${domain}.directory}/key.pem";
+              };
+              enable_https = mkOption {
+                type = types.bool;
+                description = "Whether to enable HTTPS for the web UI";
+                default = cfg.settings.misc.https_cert != null;
+                defaultText = "cfg.settings.misc.https_cert != null";
+                example = true;
+              };
+              cache_limit = mkOption {
+                type = types.str;
+                description = ''
+                  Size of the RAM cache, in bytes (prefixes supported).
+                  Sabnzbd recommends 25% of available RAM. Empty means
+                  no cache.
+                '';
+                default = "";
+                example = "500M";
+              };
+              html_login = mkOption {
+                type = types.bool;
+                description = ''
+                  Prompt for login with an html login mask if enabled,
+                  otherwise prompt for basic auth (useful for SSO)
+                '';
+                default = true;
+              };
+              inet_exposure = mkOption {
+                type =
+                  types.coercedTo
+                    (types.enum [
+                      "none"
+                      "api (add nzbs)"
+                      "api (no config)"
+                      "api (full)"
+                      "api+web (auth needed)"
+                      "api+web (locally no auth)"
+                    ])
+                    (
+                      val:
+                      if val == "none" then
+                        0
+                      else if val == "api (add nzbs)" then
+                        1
+                      else if val == "api (no config)" then
+                        2
+                      else if val == "api (full)" then
+                        3
+                      else if val == "api+web (auth needed)" then
+                        4
+                      # if val == "api+web (locally no auth)" then
+                      else
+                        5
+                    )
+                    (
+                      types.enum [
+                        0
+                        1
+                        2
+                        3
+                        4
+                        5
+                      ]
+                    );
```

What about something like this (image the correct indentation):

```suggestion
type =
let
  enum_values = {
    "none" = 0;
    "api (add nzbs)" = 1;
    "api (no config)" = 2;
    "api (full)" = 3;
    "api+web (auth needed)" = 4;
    "api+web (locally no auth)" = 5;
  };
in
types.coercedTo
  (types.enum (lib.attrNames enum_values))
  (val: enum_values.${name})
  (types.enum (lib.attrValues enum_values));
```

---

Same suggestion for other enum options. Could perhaps introduce a small helper.